src/include/cell_inline.h

```diff
@@ -205,13 +205,31 @@ __wt_cell_pack_addr(WT_SESSION_IMPL *session, WT_CELL *cell, u_int cell_type, ui
 
     __cell_pack_addr_validity(session, &p, ta);
 
+    /*
+     * If passed fast-delete information, override the cell type and append the fast-delete
+     * information after the aggregated timestamp information.
+     */
+    if (page_del != NULL) {
+        WT_ASSERT(session, cell_type == WT_CELL_ADDR_LEAF || cell_type == WT_CELL_ADDR_LEAF_NO);
+        WT_ASSERT(session,
+          page_del->prepare_state == WT_PREPARE_INIT ||
+            page_del->prepare_state == WT_PREPARE_RESOLVED);
+#ifdef WT_STANDALONE_BUILD_FT_FIX
+        WT_IGNORE_RET(__wt_vpack_uint(&p, 0, page_del->txnid));
+        WT_IGNORE_RET(__wt_vpack_uint(&p, 0, page_del->timestamp));
+        WT_IGNORE_RET(__wt_vpack_uint(&p, 0, page_del->durable_timestamp));
```

Comment:
I'm surprised these are included unconditionally - we generally only include time information if it isn't globally visible.

Reply:
Even selectively, I think we should only write the necessary time information.
Why would we include a transaction ID here, if the ID is globally visible when the record is written? It's just wasted space. Or vice versa with the timestamps.

Reply:
txnid I think is needed for rts to rollback fast truncate committed concurrently to checkpoint.

Reply:
The txnids that are performed in parallel to the checkpoint are not globally visible until the checkpoint is finished, so we don't have the problem of losing the transaction ids.

I do remember that we didn't reset the txnids and timestamps separately due to the issue described in [here](https://jira.mongodb.org/browse/WT-8074?focusedCommentId=4104281&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4104281).

But I believe we do not have to follow a similar logic for the fast truncates. I agree with Alex that we can reset the transaction ids and timestamps separately for fast truncate operations. But it needs to be commented on clearly why we do this only fast truncate time windows.

Reply:
@agorrod, I would rather not make this change for a few reasons: (1) as @kommiharibabu noted, we only write this data if the change isn't globally visible, (2) at least some of the values we don't care about are 0s and those are only a single wasted packed byte, (3) I'm nervous about this information being sufficient for future updates to the system (for example, if we were to want to flush in-progress prepared fast-truncate operations some day would the information we need change), and (4) this change has been tricky enough, and we're close enough to a deadline, that I'm not excited about comments explaining why fast-truncate is different from other operations.

If you want this change, someone else will  have to make it, I don't understand the issues well enough.

Reply:
Also note that in order to know whether the value is there when reading the cell back in, we'd need to add a byte for flags (I think *) and so the maximum possible space savings is one byte, and that only in the case where we need the transaction ID but not the timestamps. To me that seems marginal enough to not bother...

(*) Maybe there's an available flag bit in the cell header, but I think there's a reason we already didn't do that and used a page flag instead.

One thing we can do though that's pretty much free is encode the durable ts as an offset from the main ts.

Reply:
It looks like we don't encode the durable ts that way elsewhere though so maybe it's not a good idea...

Reply:
ack - thanks for exploring. I'm happy with the resolution.